plugins/embed-optimizer/class-embed-optimizer-tag-visitor.php

```diff
@@ -288,14 +288,35 @@ private function add_preconnect_links( OD_Tag_Visitor_Context $context ): void {
 		$embed_wrapper_xpath = self::get_embed_wrapper_xpath( $processor->get_xpath() );
 
 		foreach ( $this->get_preconnect_urls( $processor ) as $preconnect_url ) {
+			$host = wp_parse_url( $preconnect_url, PHP_URL_HOST );
+
+			// Skip if malformed URL.
+			if ( empty( $host ) ) {
+				continue;
+			}
+
+			// Determine if same-origin.
+			$is_same_origin = wp_parse_url( home_url(), PHP_URL_HOST ) === $host;
+
+			/**
+			 * Filter whether preconnect should be used for a given embed host.
+			 *
+			 * Allows developers to override behavior.
+			 *
+			 * @param bool   $use_preconnect Whether to use preconnect (default true for same-origin).
+			 * @param string $preconnect_url The URL being hinted.
+			 * @param string $host           The parsed host.
+			 */
+			$use_preconnect = apply_filters( 'embed_optimizer_use_preconnect', $is_same_origin, $preconnect_url, $host );
+
 			foreach ( $context->url_metric_group_collection as $group ) {
 				if ( $group->get_element_max_intersection_ratio( $embed_wrapper_xpath ) < PHP_FLOAT_EPSILON ) {
 					continue;
 				}
 
 				$context->link_collection->add_link(
 					array(
-						'rel'  => 'preconnect',
+						'rel'  => $use_preconnect ? 'preconnect' : 'dns-prefetch',
```

Comment:
Per above. Also, the phpstan type for will need to be updated to allow `dns-prefetch`.

```suggestion
						'rel'  => 'dns-prefetch',
```